controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -251,7 +251,19 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 		return r.upgradeControlPlane(ctx, cluster, kcp, ownedMachines, requireUpgrade)
 	}
 
-	// If we've made it this far, we we can assume that all ownedMachines are up to date
+	//If we've made it this far, we can assume that all owned machines are up to date
+	w, err := r.managementCluster.GetWorkloadCluster(ctx, util.ObjectKey(cluster))
+	if err != nil {
+		logger.Error(err, "failed to get remote client for workload cluster")
+	} else {
+		if w.IsWorkloadClusterUp(ctx) {
```

Comment:
This will clear up one of the linter complaints:
```suggestion
	} else if w.IsWorkloadClusterUp(ctx) {
```